src/main/java/com/minecolonies/coremod/inventory/ContainerRack.java

```diff
@@ -105,20 +106,28 @@ public ContainerRack(@NotNull final TileEntityRack tileEntityRack, @Nullable fin
         this.rack = tileEntityRack;
         this.neighborRack = neighborRack;
         this.inventorySize = this.inventory.getSlots() / INVENTORY_COLUMNS;
+        final int size = this.inventory.getSlots();
 
+        final int columns = inventorySize <= INVENTORY_BAR_SIZE ? INVENTORY_COLUMNS : ((size / INVENTORY_BAR_SIZE) + 1);
+        final int extraOffset = inventorySize <= INVENTORY_BAR_SIZE ? 0 : 2;
+        int index = 0;
 
-        for (int j = 0; j < this.inventorySize; ++j)
+        for (int j = 0; j < Math.min(this.inventorySize, INVENTORY_BAR_SIZE); ++j)
```

Comment:
Is this correct?

Reply:
Uh, let me check if it should be INVENTORY_BAR_SIZE - 1

Reply:
Nope, its correct